docs/development.md

```diff
@@ -219,6 +219,8 @@ The steps below are provided in a convenient script in [hack/create-dev-cluster.
 CLUSTER_NAME=<my-capz-cluster-name> ./hack/create-dev-cluster.sh
 ```
 
+   NOTE: `CLUSTER_NAME` can only include letters, numbers, and hyphens and can't be longer than 44 characters.
```

Comment:
Are there any user facing docs that should also be updated? Probably the CAPI book / quickstart at least.

Reply:
couldn't find a better spot in docs here, but I can add a PR in cluster-api for the quick start

Reply:
yeah let's PR that in the quick start

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/3423